test/e2e_node/system_node_critical_test.go

```diff
@@ -89,9 +89,9 @@ var _ = SIGDescribe("SystemNodeCriticalPod", framework.WithSlow(), framework.Wit
 					if hasNodeCondition(ctx, f, v1.NodeDiskPressure) {
 						return nil
 					}
-					msg := fmt.Sprintf("NodeCondition: %s not encountered yet", v1.NodeDiskPressure)
-					framework.Logf(msg)
-					return fmt.Errorf(msg)
+					msgf := "NodeCondition: %s not encountered yet"
+					framework.Logf(msgf, v1.NodeDiskPressure)
+					return fmt.Errorf(msgf, v1.NodeDiskPressure)
```

Comment:
seems kind of silly to do sprintf twice, we could just pass these to non-format errors.New() and a framework.Log ?

cc @pohly @aojea

Reply:
Agreed.

I'm not sure why the error gets logged at all. Normally, that should be handled by the caller.